core/src/scene/sceneLoader.cpp

```diff
@@ -0,0 +1,268 @@
+#include "sceneLoader.h"
+
+#include "scene.h"
+#include "tileManager.h"
+#include "view.h"
+#include "lights.h"
+#include "geoJsonSource.h"
+#include "protobufSource.h"
+
+#include "yaml-cpp/yaml.h"
+
+using namespace YAML;
+
+void loadSources(Node sources, TileManager& tileManager);
+void loadLights(Node lights, Scene& scene);
+void loadCameras(Node cameras, View& view);
+
+void SceneLoader::loadScene(const std::string& _file, Scene& _scene, TileManager& _tileManager, View& _view) {
+
+    Node config = YAML::LoadFile(_file);
+    
+    loadSources(config["sources"], _tileManager);
+    loadLights(config["lights"], _scene);
+    loadCameras(config["cameras"], _view);
+    
+}
+
+glm::vec4 parseVec4(const Node& node) {
+    glm::vec4 vec;
+    int i = 0;
+    for (auto it = node.begin(); it != node.end(); ++it) {
+        if (i < 4) {
+            vec[i++] = it->as<float>();
+        } else {
+            break;
+        }
+    }
+    return vec;
+}
+
+glm::vec3 parseVec3(const Node& node) {
+    glm::vec3 vec;
+    int i = 0;
+    for (auto it = node.begin(); it != node.end(); ++it) {
+        if (i < 3) {
+            vec[i++] = it->as<float>();
+        } else {
+            break;
+        }
+    }
+    return vec;
+}
+
+void loadSources(Node sources, TileManager& tileManager) {
+    
+    if(!sources) {
+        logMsg("Warning: No source defined in the yaml scene configuration.\n");
+        return;
+    }
+    
+    for (auto it = sources.begin(); it != sources.end(); ++it) {
+        
+        const Node source = it->second;
+        std::string type = source["type"].as<std::string>();
+        std::string url = source["url"].as<std::string>();
+        
+        std::unique_ptr<DataSource> sourcePtr;
+        
+        if (type == "GeoJSONTiles") {
+            sourcePtr = std::unique_ptr<DataSource>(new GeoJsonSource());
```

Comment:
Will "sourcePtr.reset(new GeoJsonSource())" be a better choice over the assignment operator!. 
However that being said, based on the documentation it seems they both perform the same and are more or less analogous.